# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/electricfence-uclibc.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- electric-fence-2.1.13/page.c.orig	2001-04-11 07:44:04.000000000 -0700
+++ electric-fence-2.1.13/page.c	2005-01-21 22:52:00.000000000 -0800
@@ -42,8 +42,8 @@
 #elif ( defined(_AIX) )
 	return strerror(errno);
 #else
-	if ( errno > 0 && errno < sys_nerr )
-		return sys_errlist[errno];
+	if ( errno > 0 && strerror(errno) != NULL)
+		return strerror(errno);
 	else
 		return "Unknown error.\n";
 #endif
